### PR TITLE
#29886 - Replace !r for repr() on pandas/io/parses.py

### PR DESCRIPTION
**Pull request by JvPy**
- [ ] closes #xxxx
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/io/parsers.py [hunk 1]
```diff
@@ -3600,15 +3600,15 @@ def __init__(self, f, colspecs, delimiter, comment, skiprows=None, infer_nrows=1
         self.comment = comment
         if colspecs == "infer":
             self.colspecs = self.detect_colspecs(
-                infer_nrows=infer_nrows, skiprows=skiprows
+                infer_nrows=infer_nrows, skiprows=skiprow
```
Comment (MomIsBestFriend) [inline pandas/io/parsers.py]: You accidentally deleted a ``s``` here:)

```suggestion
                infer_nrows=infer_nrows, skiprows=skiprows
```

Comment (JvPy) [reply pandas/io/parsers.py]: I did, sry :(
I made another PR with the fix, but it failed

Ps: This is my first time contributing!

File: pandas/io/parsers.py [hunk 2]
```diff
@@ -3607,8 +3607,8 @@ def __init__(self, f, colspecs, delimiter, comment, skiprows=None, infer_nrows=1
 
         if not isinstance(self.colspecs, (tuple, list)):
             raise TypeError(
-                "column specifications must be a list or tuple, "
-                "input was a %r" % type(colspecs).__name__
+                f"column specifications must be a list or tuple, "
+                f"input was a {repr(type(colspecs).__name__)}"
```
Comment (jreback) [inline pandas/io/parsers.py]: you don't need any of the repr except for this one as they are all repr'ing strings.

**Review by jreback — CHANGES_REQUESTED** [review]
you don't ne

File: pandas/io/parsers.py (hunk 2 above)
Comment (JvPy) [reply pandas/io/parsers.py]: Show I leave it like this? 
f"input was a {type(colspecs).__name__}"

Comment (JvPy) [reply pandas/io/parsers.py]: Should I change the other strings to something like this?

f"input was a {(type(colspecs).__name__)}"

**Comment by JvPy** [comment]
@jreback build fails if I remove repr. Can you help me with this issue?

File: pandas/io/parsers.py (hunk 2 above)
Comment (jreback) [reply pandas/io/parsers.py]: yesp (but you can leave anything that is not a string with ``repr(...)``, e.g. this one is a list

**Review by jreback — CHANGES_REQUESTED** [review]
also pls rebase

File: pandas/io/parsers.py [hunk 3]
```diff
@@ -3607,8 +3607,8 @@ def __init__(self, f, colspecs, delimiter, comment, skiprows=None, infer_nrows=1
 
         if not isinstance(self.colspecs, (tuple, list)):
             raise TypeError(
-                "column specifications must be a list or tuple, "
-                "input was a %r" % type(colspecs).__name__
+                f"column specifications must be a list or tuple, "
```
Comment (WillAyd) [inline pandas/io/parsers.py]:
```suggestion
                "column specifications must be a list or tuple, "
```

The `f` prefix is only needed if actually substituting

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/io/parsers.py (hunk 3 above)
Comment (JvPy) [reply pandas/io/parsers.py]: Noted. I'll change it tonight!

File: pandas/io/parsers.py [hunk 4]
```diff
@@ -1080,7 +1080,7 @@ def _clean_options(self, options, engine):
             if not isinstance(converters, dict):
                 raise TypeError(
                     f"Type converters must be a dict or subclass, "
```
Comment (WillAyd) [inline pandas/io/parsers.py]:
```suggestion
                    "Type converters must be a dict or subclass, "
```

I think I missed this in last review if you want to accept commit on GitHub

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/io/parsers.py (hunk 4 above)
Comment (WillAyd) [reply pandas/io/parsers.py]: Ah actually I see you haven't touched this, but OK to clean up here

Comment (JvPy) [reply pandas/io/parsers.py]: Ill fix that for you, no problem!

**Review by WillAyd — APPROVED** [review]
lgtm @jreback

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks

**Pull request merged**